### PR TITLE
Fix #9982 dropdown style and toc configuration improvements

### LINKED ISSUE #9982: Alternative layouts for the TOC
## Description

This issue collects all the needed changes to review the table of content:

## Acceptance criteria

- [ ] Refactor of TOC components, plugin and styling
- [ ] Refactor of drag and drop of components
- [ ] Refactor of filtering system
- [ ] Improve title visibility

## Other useful information

```[tasklist]
### Task to take into account in the development
- [ ] #8918
- [ ] #9430
- [ ] #8953
- [ ] #8471
- [ ] #4255
- [ ] https://github.com/GeoNode/geonode-mapstore-client/issues/1265
```

### PR DESCRIPTION
**Pull request by allyoucanmap**
## Description

Fixes for https://github.com/geosolutions-it/MapStore2/pull/9983#issuecomment-2015107021

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix

## Issue

**What is the current behavior?**

#9982

**What is the new behavior?**

The theme for dropdown options inside toc has been fixed, the swipe tool correctly apply only in 2D maps and the embedded configuration reviewed.

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)

 - [x] No

### CODE REVIEW

File: web/client/hooks/__tests__/usePluginItems-test.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2022, GeoSolutions Sas.
```
Comment (allyoucanmap) [inline web/client/hooks/__tests__/usePluginItems-test.js:2]:
```suggestion
 * Copyright 2024, GeoSolutions Sas.
```

**Review by offtherailz — APPROVED** [review]
LGTM. I tested the menu and looked at the code changes, that clook consistent.
I didn't tested all the points, I let you test them @ElenaGallo

**Comment by offtherailz** [comment]
@ElenaGallo, could you please test this on  DEV  ? Thank you

**Comment by ElenaGallo** [comment]
Test passed, @allyoucanmap please backport to 2024.01.xx. Thanks

**Pull request merged**